python/tank/platform/application.py

```diff
@@ -135,7 +151,26 @@ def destroy_app(self):
         Called by the engine as it is being destroyed.
         """
         pass
-    
+
+    def pre_context_change(self, old_context, new_context):
```

Comment:
Should this be on the bundle level? What if an loaded engine or framework needs to update an internal state due to context switch? Or maybe I am overthinking this?